### ISSUE TITLE
Potential bug in insert waypoint code

### ISSUE THREAD

**Issue by sergei1152** [UWARG/PICpilot#79]
The struct for waypoint wrapper looks like this:
```c
//Structs and typedefs
typedef struct _waypointWrapper{
    long double longitude;  //TODO: Longitude and Latitude is bulky. If problems arise, change the format.
    long double latitude;
    float altitude;
    float radius; //Radius of turn
    char type; //Regular or probe drop location
    char id;    //Array ID
    char nextId; //For use with insertNode() or operations that require reference to another node
    char previousId; //For use with insertNode() or operations that require reference to another node
}WaypointWrapper;
```

The groundstation/data relay station will send latitude,longitude,altitude,radius,type,next_id, previous_id

It however does not send the `id` field, which will cause the parsing of the nextId and previousId values on the picpilot side to fail, as the code relies on the exact structure that the struct is defined in
```c
 Picpilot code:
case INSERT_WAYPOINT:
                amData.waypoint.altitude = (*(WaypointWrapper*)(&cmd->data)).altitude;
                amData.waypoint.latitude = (*(WaypointWrapper*)(&cmd->data)).latitude;
                amData.waypoint.longitude = (*(WaypointWrapper*)(&cmd->data)).longitude;
                amData.waypoint.radius = (*(WaypointWrapper*)(&cmd->data)).radius;
                amData.waypoint.nextId = (*(WaypointWrapper*)(&cmd->data)).nextId;
                amData.waypoint.previousId = (*(WaypointWrapper*)(&cmd->data)).previousId;
                amData.waypoint.type = (*(WaypointWrapper*)(&cmd->data)).type;
                amData.command = PM_INSERT_WAYPOINT;
                amData.checkbyteDMA = generateAMDataDMACheckbyte();
                amData.checksum = generateAMDataChecksum(&amData);
                break;
```
The code should be refactored slightly such that the fields that are used across the UPDATE_WAYPOINT, NEW_WAYPOINT, INSERT_WAYPOINT, etc commands are placed first in the struct, with padding bytes on the data relay where necessary to avoid parse errors.

To start off, the id field should be moved to the bottom of the struct, since its never actually used. It should be notes in the comments that the structure of the struct is vital, so that people dont insert fields in the middle of it.

**Comment by ianjfrosst** [comment]
Yeah, this is definitely a problem. I'm working on this on [my own fork](https://github.com/ianjfrosst/PICpilot/tree/multipart_commands), as well as a fork of [data-relay](https://github.com/ianjfrosst/data-relay-station/tree/multipart_commands).

**Comment by ianjfrosst** [comment]
Fixed in PR #81
